python-package/lightgbm/dask.py

```diff
@@ -102,7 +107,7 @@ def _find_ports_for_workers(client: Client, worker_addresses: Iterable[str], loc
     return worker_ip_to_port
 
 
-def _concat(seq):
+def _concat(seq: Iterable[_DaskPart]) -> _DaskPart:
```

Comment:
I didn't have any special reason for choosing `Iterable` instead of `List`. I was just looking at the code and didn't see any list-specific stuff so I thought it made sense to use the more general thing.

But since this is a totally-internally function, where we control the input, and since `Iterable` does weird stuff as that S/O post points out, I'll change this to `List[]`

Reply:
fixed in https://github.com/microsoft/LightGBM/pull/3866/commits/bfd9dc0a7438a45901afe916c095bfb818b1dba9